packages/providers/mgt-teamsfx-provider/src/TeamsFxProvider.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { IProvider, ProviderState, createFromProvider } from '@microsoft/mgt-element';
+import { TeamsUserCredential } from '@microsoft/teamsfx';
+
+/**
+ * TeamsFx Provider handler
+ *
+ * @export
+ * @class TeamsFxProvider
+ * @extends {IProvider}
+ */
+export class TeamsFxProvider extends IProvider {
+  /**
+   * Name used for analytics
+   *
+   * @readonly
+   * @memberof TeamsFxProvider
+   */
+  public get name(): string {
+    return 'MgtTeamsFxProvider';
+  }
+
+  /**
+   * returns _credential
+   *
+   * @readonly
+   * @memberof TeamsFxProvider
+   */
+  public get credential(): TeamsUserCredential {
+    return this._credential;
+  }
+
+  /**
+   * Privilege level for authentication
+   *
+   * @type {string[]}
+   * @memberof TeamsFxProvider
+   */
+  private scopes: string[] = [];
```

Comment:
```suggestion
  private readonly scopes: string[] = [];
```

Reply:
We have the ability to update the scopes while the user is using the app (think contextual consent)